FF-152-add-my-profile-page

Ksenialiashchuk opened the pull request:
Added page profile mobi with input validation and custom calendar.
<img width="406" height="578" alt="Снимок экрана 2025-07-24 в 17 58 44" src="https://github.com/user-attachments/assets/09f47772-92a5-4769-b7c7-61bbcc806c90" />
<img width="406" height="578" alt="Снимок экрана 2025-07-24 в 17 59 03" src="https://github.com/user-attachments/assets/3b81a12b-8442-470d-858d-c06dee37be74" />
<img width="406" height="578" alt="Снимок экрана 2025-07-24 в 17 59 28" src="https://github.com/user-attachments/assets/c22f97a6-c88b-46f0-a0c4-3a0f9a75fed4" />
<img width="406" height="578" alt="Снимок экрана 2025-07-24 в 17 59 40" src="https://github.com/user-attachments/assets/3ef3101e-a7cb-48f5-a3a9-869c5e8a760e" />
<img width="406" height="578" alt="Снимок экрана 2025-07-24 в 17 59 55" src="https://github.com/user-attachments/assets/56d337db-44ba-4a30-af19-54df452354ff" />
<img width="406" height="578" alt="Снимок экрана 2025-07-24 в 18 00 11" src="https://github.com/user-attachments/assets/9ca289d8-b2c1-4fe6-972d-1f108b7bcec0" />

Ksenialiashchuk committed:
FF-152-add-my-profile-page

Ksenialiashchuk committed:
FF-152-fix-code-in-calendarProfile

Ksenialiashchuk committed:
FF-152-fix-code-for-build

Margo-Bond commented on frontend-react/components/mobi/shared/CalendarProfileMobi/CalendarProfileMobi.tsx:
```diff
@@ -0,0 +1,230 @@
+        years.indexOf(selectedDate.getFullYear())
+    );
+
+    useEffect(() => {
+        const newDate = new Date(
+            years[visibleYearIndex],
+            visibleMonthIndex,
+            Math.min(
+                days[visibleDayIndex],
+                new Date(years[visibleYearIndex], visibleMonthIndex + 1, 0).getDate()
+            )
+        );
+        setSelectedDate(newDate);
+    }, [visibleDayIndex, visibleMonthIndex, visibleYearIndex]);
```
useEffect обновляет selectedDate при любой смене индекса, даже если дата не изменилась, добавь условие

Margo-Bond commented on frontend-react/components/mobi/shared/CalendarProfileMobi/CalendarProfileMobi.tsx:
```diff
@@ -0,0 +1,230 @@
+    onCancel: () => void;
+};
+
+const DatePickerCalendar: React.FC<DatePickerCalendarProps> = ({
```
Все компоненты должны иметь суффикс Desktop/Mobi (README)

Margo-Bond commented on frontend-react/components/mobi/shared/CalendarProfileMobi/react-mobile-datepicker.d.ts:
Почему не установила компонент react-mobile-datepicker в проект?

Margo-Bond commented on frontend-react/components/ui/calendarProfile.tsx:
Для чего устанавливаешь компонент из schadcn CalendarProfile, если нигде его не используешь?

Margo-Bond commented on frontend-react/modals/ModalsDesktop/ModalCookieDesktop.tsx:
пропс onClose является общим для всего проекта, не нужно его переименовывать

Margo-Bond commented on frontend-react/modals/modals.tsx:
Вернуть имя пропса

Margo-Bond commented on frontend-react/styles/customClasses/customClassElements.css:
Тут верни цвет какой был, поскольку этот класс применяется в других компонентах

Margo-Bond reviewed (commented):
1. Название ветки не соответствует правилам, указанным в README. Необходимо переименовать.
Имя ветки должно быть `FF-152-feat/my-profile-page-mobi`, а заголовок PR был ошибочно оформлен в стиле имени ветки, должно быть без дефисов, например **FF-152-feat my profile page for mobi**
2. Неверно именованы коммиты, в README указана ссылка на общие правила именования коммитов.
Сейчас переименовывать не надо.
3. Не указано описание ПР и не добавлены скриншоты для всех необходимых мобильных медиа запросов. В README также эта информация указана, добавить в соответствии с правилами.
4. Изучить файл README

Ksenialiashchuk committed:
FF-152 fix: resolve code review comments (part)

Ksenialiashchuk committed:
FF-152-fix-page-style

Ksenialiashchuk committed:
FF-152 feat(profile): add profile logic for mobile

Added user profile page for the mobile version.
Implemented calendar component and input validation.

Ksenialiashchuk committed:
FF-152 feat(profile): add profile logic for mobile

Added input logic for submit button. Fixed page style.

Ksenialiashchuk committed:
FF-152-fix: fix conflicts

Ksenialiashchuk committed:
FF-152-fix: fix conflicts

Ksenialiashchuk committed:
FF-152-fix: fix calendar component

Ksenialiashchuk committed:
Merge branch 'refs/heads/dev-front-react' into FF-152

Ksenialiashchuk commented:
не могу воспроизвести @ILMPI 

_Размер placeholder-а и ввода текста._

Ksenialiashchuk committed:
FF-152-fix: fix profile logic for mobile

Fixed comments: fixed input styles, calendar style and validation logic.

Ksenialiashchuk committed:
FF-152-fix: fix code

Fixed interface IPhoneInputMobi

Ksenialiashchuk committed:
Merge branch 'refs/heads/dev-front-react' into FF-152

Ksenialiashchuk committed:
FF-152-fix: fix navigation menu

Ksenialiashchuk commented:
<img width="250" height="594" alt="Снимок экрана 2025-08-05 в 17 17 37" src="https://github.com/user-attachments/assets/da0c9f46-c7b7-4aab-819c-656722fff80f" />
<img width="250" height="68" alt="Снимок экрана 2025-08-05 в 17 18 28" src="https://github.com/user-attachments/assets/9dd15dd8-736d-465e-b147-e9ede6d2709b" />
<img width="250" height="68" alt="Снимок экрана 2025-08-05 в 17 18 40" src="https://github.com/user-attachments/assets/25456cfe-c1c3-453e-b3c7-37e0c4602d51" />
<img width="250" height="184" alt="Снимок экрана 2025-08-05 в 17 18 58" src="https://github.com/user-attachments/assets/d4e7009b-078a-46d2-8281-c4e635cd4cbf" />

Fixed profile page mobi

Ksenialiashchuk committed:
FF-152-fix: fix MyProfile page mobi

Ksenialiashchuk committed:
Merge branch 'dev-front-react' into FF-152

Ksenialiashchuk committed:
FF-152-fix: fix build

Ksenialiashchuk commented:
исправлены комментарии по коду. 
-----
появилась возможность закрывать списки за их пределами.
-----
уточнила моменты по инпуту с номером телефона, все осталось, как было

ILMPI commented:
> > не могу воспроизвести @ILMPI
> > _Размер placeholder-а и ввода текста._
> 
> На разрешение sm_l и меньше
> 
> <img alt="image" width="476" height="881" src="https://private-user-images.githubusercontent.com/82223578/475196054-4062dd7c-b725-462f-ac70-1ceb74297a4c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgyMzY0MDQsIm5iZiI6MTc1ODIzNjEwNCwicGF0aCI6Ii84MjIyMzU3OC80NzUxOTYwNTQtNDA2MmRkN2MtYjcyNS00NjJmLWFjNzAtMWNlYjc0Mjk3YTRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE4VDIyNTUwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlkNTJhNGJiYzgzYzY5NjhjMzI0OTRlY2MzYmM1MzY5OTA0MjVkMjVkY2U2NmJlYmI2ODEzYzBkMDU4NzMzNmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RrtIcgqOHEzoKtfVUAyzGpH4ELeriv8tJVgxXycxUj8">

не исправлено

Ksenialiashchuk committed:
FF-152-fix: fixed input styles

Ksenialiashchuk commented:
поправила комментарии и стили

ILMPI commented:
- _Единый формат заполненного поля._

- Сейчас при заполнении поля Даты рождения цвет шрифта не меняется.

<img width="349" height="271" alt="image" src="https://github.com/user-attachments/assets/d041e24e-363d-4d84-8b40-34de151e3607" />

- Формат заполненного поля в дизайне выглядит вот так (если есть какие-то изменения по этому вопросу уточни пожалуйста, если нет то не хватает изменения фона поля):
<img width="349" height="91" alt="image" src="https://github.com/user-attachments/assets/13e1ceb5-f0ff-4b90-8bf0-49a79b111ec7" />

- _Поле с выпадающим списком не закрывается без выбора значения._ Если нажать на поле а затем перейти к редактированию другого поля список так и остается открытым.
<img width="347" height="521" alt="image" src="https://github.com/user-attachments/assets/ee7e728b-46f9-4f9f-bd75-4c5a12806eae" />

- Также не меняется бордер при выборе активного поля.

- _Исправить размер шрифта плейсхолдера телефона для разрешения экрана от 375 и меньше._

<img width="785" height="195" alt="image" src="https://github.com/user-attachments/assets/1d01c878-efd4-449c-b85a-929d8f4f745e" />

Ksenialiashchuk committed:
FF-152-fix: fixed input styles

fixed validation styles except phone input and calendar
